File: VERSION PYTHON/RockPaperScissor.py
import time
import random

Plays = ["Rock", "Paper", "Scissor"]
Counters = {
    "Rock": "Paper",
    "Paper": "Scissor",
    "Scissor": "Rock"
}
Nums =  ["0","1","2","3","4","5","6","7","8","9"]

def InfiniteMode(IsScore, Inf, Per):
  GameScore = 0
  YourScore = 0
  I = 1
  if Inf and not Per:
    Inf = 100000000000000000
  elif not Inf and not Per:
    Inf = 10
    
  while I <= Inf:
    I+=1
    UserChoice = input(f"Type your Choice between {Plays}").capitalize()
    GameChoice = random.choice(Plays)
    
    if UserChoice not in Plays:
      print("Wrong Choice")
      InfiniteMode(IsScore, Inf, Per)
  
    print("The game is thinking....")
    time.sleep(2)
    print(f"Game choice: {GameChoice}")
    time.sleep(1)
    print(f"Your choice: {UserChoice}")
    time.sleep(1)
  
    if UserChoice == GameChoice:
      print("Tie")
      GameScore += 1 if IsScore else GameScore
      YourScore += 1 if IsScore else YourScore
    elif UserChoice in Counters[GameChoice]:
      print("You Win")
      YourScore += 1 if IsScore else YourScore
    else:
      print("You Lose")
      GameScore += 1 if IsScore else GameScore
      
    print(f"Your Score: {YourScore}" if IsScore else "")
    print(f"Game Score: {GameScore}" if IsScore else "")
      
def Main():
  print("Welcome to the Rock Paper Scissor Game")
  time.sleep(1)
  print("What mode do you want to play?")
  print("1. Infinite Mode without score system")
  print("2. Infinite Mode with score system")
  print("3. 10 rounds without scoring")
  print("4. 10 rounds with scoring")
  print("5. Customized Mode")
  Choice = int(input("Your Choice: "))
  if Choice == 1:
    InfiniteMode(False, True, False)
  elif Choice == 2:
    InfiniteMode(True, True, False)
  elif Choice == 3:
    InfiniteMode(False, False, False)
  elif Choice == 4:
    InfiniteMode(True, False, False)
  elif Choice == 5:
    Score = False
    rounds = 0
    print("How many rounds do you want?")
    print("You can say infinity to infinity rounds")
    time.sleep(1)
    Escolha = input("Here: ")
    if str(Escolha).lower() == "infinity":
      rounds = 100000000000000000
    elif "infinity" not in Escolha.lower() and not any(num in Escolha for num in Nums):
        print("Wrong Choice")
        Main()
    else:
      rounds = int(Escolha)
    time.sleep(1)
    print("Do you want score system?")
    time.sleep(1)
    print("1. Yes")
    time.sleep(1)
    print("2. No")
    Escolha = input("Here: ")
    if int(Escolha) == 1:
      Score = True
    
    InfiniteMode(Score, rounds, True)
  else:
    print("Invalid Choice")
    Main()
    
Main()
#TODO: SEGURANÇA NOS INPUTS

File: VERSION JAVASCRIPT/RockPaperScissor.js
const Plays = ["Rock", "Paper", "Scissor"];
const Modes = ["1. Infinity Mode", "2. Infinity Mode with score", "3. 10 Rounds without score", "4. 10 rounds with score", "5. Personalized mode"];
const Counters = {
  Rock: "Paper",
  Paper: "Scissor",
  Scissor: "Rock"
};

let Winner;
let Total = 4;
let current = 0;
let UserScore = 0;
let GameScore = 0;
let NRound = 0;

function Finish() {
  console.log("Game Finished");
}

function SelectMode() {
let Mode = prompt(`Escolha um modo, entre "${Modes}", type E to exit te game`);

if (Mode === "1") {
  InfinityMode(false, 1000000000000, false, 0)
} else if (Mode === "2") {
  InfinityMode(true, 1000000000000, false, 0)
} else if (Mode === "3") {
  InfinityMode(false, 1, false, 0)
} else if (Mode === "4") {
  InfinityMode(true, 1, false, 0)
} else if (Mode === "5") {
  Nums = Number(prompt("Quantos rounds vc quer?, type -1 for inf"));
  if (Nums === -1) {
    Nums = 1000000000000;
  }

  Scr = prompt("Quer score? (s/n)?");
  if (Scr.toLowerCase() === "s") {
    mds = prompt("deseja encerrar o jogo quando tiver pontuação específica? (s/n)");
    if (mds.toLowerCase() === "s") {
      TotP = Number(prompt("Qual é a pontuaçáo?"));
      InfinityMode(true, Nums, true, TotP)
    } else {
      InfinityMode(true, Nums, false, 0)
    } 
  } else {
    InfinityMode(false, Nums, false, 0)
  }
} else if(Mode === "E") {
  Finish()
} else {
  alert("Invalid Choice");
  SelectMode()
  return;
}
}


function InfinityMode(Score, HowR, Spef, QSpef) {
  
  let UserChoice = prompt(`Sua jogada, entre ${Plays}, type R to return back`);
  let GameChoice = Plays[Math.floor(Math.random() * 3)];
  
  if (UserChoice === Counters[GameChoice]) {
    Winner = "You Win";
    UserScore++;
  } else if (UserChoice === GameChoice) {
    Winner = "Tie";
    UserScore++;
    GameScore++;
  } else if(UserChoice.toUpperCase() === "R") {
    UserScore = 0;
    GameScore = 0;
    NRound = 0;
    SelectMode()
  } else {
    Winner = "Game win";
    GameScore++;
  }
  
  let Messages = ["The game is thinking...", `Game Choice: ${GameChoice}`, `Your choice: ${UserChoice}`, `${Winner}`];
  
  if (Score) {
    if (Score < 6) {
    Total = 6;
    }
    Messages.push(`Your Score: ${UserScore}`);
    Messages.push(`Game Score: ${GameScore}`);
  }
  
  let X = setInterval(() => {
    console.log(Messages[current]);
    current++;
    if (current === Total) {
      NRound++;
      current = 0;
      clearInterval(X);
      if (NRound === HowR) {
      Finish()
      return;
    }
    if (Spef === true && UserScore === QSpef || GameScore === QSpef) {
      Finish()
      return;
    }
      let Y = setInterval(() => {
        Choice = prompt("Another Round? 1.Yes 2.No")
        if (Choice !== undefined) {
          clearInterval(Y);
        }
      if (Choice === "1") {
        InfinityMode(Score, HowR, Spef, QSpef)
      } else if (Choice === "2") {
        SelectMode()
      }
      }, 1500);
    }
  }, 1250);
}

SelectMode()